### ISSUE TITLE
[Bug] Outputs are different with the same generator

### ISSUE THREAD

**Issue by ZDJeffrey** [mit-han-lab/nunchaku#294]
### Checklist

- [x] 1. I have searched for related issues and FAQs (https://github.com/mit-han-lab/nunchaku/discussions/262) but was unable to find a solution.
- [x] 2. The issue persists in the latest version.
- [x] 3. Please note that without environment information and a minimal reproducible example, it will be difficult for us to reproduce and address the issue, which may delay our response.
- [x] 4. If your report is a question rather than a bug, please submit it as a discussion at https://github.com/mit-han-lab/nunchaku/discussions/new/choose. Otherwise, this issue will be closed.
- [x] 5. If this is related to ComfyUI, please report it at https://github.com/mit-han-lab/ComfyUI-nunchaku/issues.
- [x] 6. I will do my best to describe the issue in English.

### Describe the Bug

When running flux.1-dev and flux.1-schnell, the output images are still slightly different when using the same generator. I tried saving the transformer input and output tensors and found that the same latent input will get different noise_pred.
I am not sure if this is because of the unset random seed in the Transformer inference code.
| | | | |
|:-:|:-:|:-:|:-:|
|First Exec|![Image](https://github.com/user-attachments/assets/e1cdd3df-9d57-40b8-8d50-d4891cc5981e)|![Image](https://github.com/user-attachments/assets/ac0a6663-71fb-44d3-a444-baab6a80f6a2)|![Image](https://github.com/user-attachments/assets/131bd8ee-f1b8-4376-bbbc-3c1b22431df0)|
|Second Exec|![Image](https://github.com/user-attachments/assets/24f7f8ec-9d32-4de3-94e0-4f3854165502)|![Image](https://github.com/user-attachments/assets/0811f175-575a-470a-acab-a529e93d1432)|![Image](https://github.com/user-attachments/assets/28107d71-a4c0-45dc-8114-49af137aa426)|

### Environment

- GPU: RTX 4090
- python: 3.10.6
- pytorch: 2.6
- CUDA: 12.4
- Nunchaku: 0.2.0

### Reproduction Steps

```python
from nunchaku import NunchakuFluxTransformer2dModel
from nunchaku.caching.diffusers_adapters import apply_cache_on_pipe
from diffusers import FluxPipeline
import torch
import numpy
import random

transformer = NunchakuFluxTransformer2dModel.from_pretrained("/file_system/models/dit/svdq-int4-flux.1-schnell")
transformer.set_attention_impl('nunchaku-fp16')

pipe = FluxPipeline.from_pretrained("/file_system/models/dit/FLUX.1-schnell", transformer=transformer, torch_dtype=torch.bfloat16).to("cuda")

prompt = "A cat holding a sign that says hello world"
seed = 42

torch.manual_seed(seed)
numpy.random.seed(seed)
random.seed(seed)


for i in range(3):
    image = pipe(
        prompt=prompt,
        height=1024,
        width=1024,
        num_inference_steps=4,
        generator=torch.Generator(device=f"cuda").manual_seed(seed),
    ).images[0]
    image.save(f"test{i}.png")
```

**Comment by lmxyy** [comment]
Hi, this is minor random precision noise caused by the GPU's accumulation order. Due to out-of-order execution, slight variations can occur in the results. Fixing the accumulation order would impact performance (i.e., speed), so we currently have no plans to change it.
